src/main/io/serial_cli.c

```diff
@@ -2800,6 +2833,34 @@ static void printConfig(char *cmdline, bool doDiff)
     }
 
     createDefaultConfig(&defaultConfig);
+    // make copies of configs to do differencing
```

Comment:
This looks good to me. As you say, it is not the world's most elegant solution, but it is a workable way to convert `diff` to parameter groups, and we can always improve it later. The way profiles are switched around for `dump all` / `diff all` is very unelegant too, and always has been that way.